src/components/SpeciesTable.js

```diff
@@ -374,7 +374,12 @@ const EditModal = ({
   data,
 }) => {
   const [error, setError] = React.useState(undefined);
-  const nameSpecies = data.map((species) => species.name.toLowerCase());
+  let objSpecies = {};
+  data.map((species) => {
```

Comment:
Populating `objSpecies` seems like unnecessary complexity.
You could just filter the species list here and leave the rest of the code as it is – maybe something like:
```
const otherSpeciesList = isEdit
  ?  data.filter((species) => Number(species.id) !== speciesEdit.id)
  :  data;
const nameSpecies = otherSpeciesList.map((species) => species.name.toLowerCase());
```